src/vos/vos_obj_index.c

```diff
@@ -865,6 +867,18 @@ oi_iter_aggregate(daos_handle_t ih, bool range_discard)
 	obj = (struct vos_obj_df *)rec_iov.iov_buf;
 	oid = obj->vo_id;
 
+	rc = vos_obj_hold(vos_obj_cache_current(cont->vc_pool->vp_sysdb), cont, oid,
+			  &oiter->oit_epr, iter->it_bound, base_flag | VOS_OBJ_NO_HOLD,
+			  DAOS_INTENT_PURGE, NULL, NULL);
```

Comment:
I don't quite see why this extra vos_obj_hold() call (with this VOS_OBJ_NO_HOLD flag) is required here, it looks to me it's in aggregation post callback with the object held, and the conflict checking has already been done in iter_prepare, did I miss anything?

Reply:
For the object iterator, we don't actually call vos_obj_hold.  It's called only in the nested dkey iterator but that's too late.

Reply:
I see, thanks.